app/javascript/components/widgets/weather/sky-icons.jsx

```diff
@@ -11,25 +11,60 @@ import CloudyIcon from '../../../../assets/images/icons/icon-1-cloud.svg';
 import PartlyCloudyDayIcon from '../../../../assets/images/icons/icon-2-cloud-sun.svg';
 import PartlyCloudyNightIcon from '../../../../assets/images/icons/icon-3-cloud-moon.svg';
 
-const iconMapping = {
-  'clear-day': <img src={ClearDayIcon} alt="clear-day" />,
-  'clear-night': <img src={ClearNightIcon} alt="clear-night" />,
-  rain: <img src={RainIcon} alt="rain" />,
-  snow: <img src={SnowIcon} alt="snow" />,
-  sleet: <img src={SleetIcon} alt="sleet" />,
-  wind: <img src={WindIcon} alt="wind" />,
-  fog: <img src={FogIcon} alt="fog" />,
-  cloudy: <img src={CloudyIcon} alt="cloudy" />,
-  'partly-cloudy-day': (
-    <img src={PartlyCloudyDayIcon} alt="partly-cloudy-day" />
-  ),
-  'partly-cloudy-night': (
-    <img src={PartlyCloudyNightIcon} alt="partly-cloudy-night" />
-  ),
+export const SkyIcon = ({ icon, width, height }) => {
+  switch (icon) {
+    case 'clear-day':
+      return (
+        <img src={ClearDayIcon} alt="clear-day" width={width} height={height} />
```

Comment:
There some duplication here. Maybe a mapping of the `icon` prop to the `src` with a final return of:
```
<img src={iconMapping[icon]} alt={icon} width={width} height={height} />
```